### ISSUE TITLE
Nextcloud support for MyOwnNotes 2?

### ISSUE THREAD

**Issue by gardar** [aykit/MyOwnNotes#140]
It would be good if we could use the nextcloud app instead of the owncloud app for authentication with MyOwnNotes 2.

**Comment by ramschmaerchen** [comment]
We decided to support ownCloud, see https://github.com/aykit/MyOwnNotes#what-do-you-think-about-nextcloud

What efforts would be needed to port for nextCloud?

**Comment by gardar** [comment]
Understood.

I'm not quite sure what efforts would be needed, the nextcloud and owncloud servers/clients are quite similar at this point and almost everything I've tried on owncloud works on nextcloud and vice versa.

I have a nextcloud 10.0.1 server and it works fine with the owncloud android app and MyOwnNotes 2 through the owncloud android app.
It would be good if I could use the nextcloud app instead of the owncloud app. I'm assuming their API integration is the same or at least really similar.

**Comment by jancborchardt** [comment]
@ramschmaerchen just for reference on the GSoC acceptance: of course they were accepted – because in all the recent years @jospoortvliet, me and others worked hard on that – and now switched to Nextcloud. Nextcloud though is in the eyes of Google a »new« project, even though most developers switched. It sucks but we will work to get accepted again. :)

We for example:
- we’re accepted to https://railsgirlssummerofcode.org
- we’re also mentoring at the regular [Ladies that FOSS](https://www.wikimedia.de/wiki/LadiesthatFOSS/Meetup) meetups
- have much more active community meetups: https://www.meetup.com/topics/nextcloud/
- have a conference coming up in Berlin https://nextcloud.com/conf/ – to which you are invited! :)
- we have the active community on Github: https://github.com/nextcloud/
- and in the forum: https://help.nextcloud.com/
- last but not least, and most relevant for you: The Notes app development is done at us https://github.com/nextcloud/notes/

So we’d love to continue working together. :) It’s mostly just a name change.
I’m sure the awesome folks of our Android community @AndyScherzinger @tobiasKaminsky @mario @David-Development would love to help if you have questions. :)

**Comment by ramschmaerchen** [comment]
@jancborchardt MyOwnNotes is a hobby project. If supporting nextCloud requires much effort we won't do it.
However, if the integration of nextCloud and ownCloud android clients is the same I guess it may be doable. If the dev of the nextCloud client could shed light on this, it would be much appreciated.

Just fyi: MyOwnNotes 2 does not depend on the server side notes application anymore.

**Comment by jancborchardt** [comment]
@ramschmaerchen sure, that’s why we’re here to help you. :) I see no reason why MyOwnNotes shouldn’t work with Nextcloud. You can actually just directly try with the demo server at https://demo.nextcloud.com

> MyOwnNotes 2 does not depend on the server side notes application anymore.

Cool! :) Even simpler.

**Comment by ramschmaerchen** [comment]
Ok, if I understand correctly, we can interact with the [nextCloud client](https://f-droid.org/wiki/page/com.nextcloud.client) the same way as we do with the ownCloud client? Do you plan to make nextCloud available on F-Droid again, see [this isse](https://github.com/nextcloud/android/commit/e245b1f91dd30994f182892255d963a62dc6429c#commitcomment-21381828).

Thanks for the hint with the test instances.

**Comment by 9662** [comment]
> We decided to support ownCloud, see https://github.com/aykit/MyOwnNotes#what-do-you-think-about-nextcloud

As a user without involvement in either project, ownCloud is dead. For us, the last drop was back in [November 2016](https://github.com/owncloud/owncloud.org/issues/1106).

Please consider switching to Nextcloud, cheers. 👍

**Comment by pecuna** [comment]
Nextcloud seems to be preferred over Owncloud as it is more actively developed and is completely opensource and free.
I am forced to use the old version of MyOwnNotes until Nextcloud app is supported.
I am sure the count of people like me will grow.

**Comment by Albirew** [comment]
Actually, version 1.6 was working fine with my Nextcloud server. Version 2.0 broke compatibility by requesting owncloud app instead of previously used credentials.

**Comment by mario** [comment]
@ramschmaerchen we already have a nightly version on F-droid without GCM, though we can't guarantee for the bugs that removing GCM brought.

https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta

I didn't check the code, but why wouldn't it work with the NC app as well? :)

Anyway, happy to help in whatever way I can.

(Just for my information: if you don't depend on the notes app, do you manually create the text files on the server?)

**Comment by ramschmaerchen** [comment]
Here my honest opinions to all of you:

OwnCloud was never known for stable releases. Bugfixing has always been a problem in the owncloud world. Seeing all those that were unable to maintain ownCloud in an enterprise way depart to nextCloud does not want me to use nextCloud.

Calling a gsoc attender dead is just not honest.

Fully free != Shipping with GCM per default.

We will look into supporting nextCloud anyways.

**Comment by jancborchardt** [comment]
@ramschmaerchen from what I can see, I’ve been the person from the ownCloud community who communicated most with you, used the app, reported issues etc. I’m with Nextcloud now. That should already be an indicator which community is really active …

> OwnCloud was never known for stable releases. Bugfixing has always been a problem in the owncloud world.

That’s some of the biggest reasons we forked and focused on more stability and security.

Simply compare this: https://github.com/owncloud/core/graphs/contributors with this: https://github.com/nextcloud/server/graphs/contributors – 8 of the top 10 contributors moved to Nextcloud.

**Comment by ramschmaerchen** [comment]
You are right and I really appreciate your effort. 

Regarding both cloud applications I don't care about either of them. I want files being backupped and accessible online. I don't care which application is doing that. OwnCloud did this so so, but Seafile is not good either so I decided to stay. If nextCloud achieves a *stable* ownCloud, I'd be happy to use it.

Regarding myownnotes support it depends on the effort needed. We will look into it.

**Comment by thomascrisan** [comment]
@pecuna I am in the same predicament of not being able to use MyOwnNotes version 2.0.  I switched to Nextcloud last year and have found the experience less frustrating than when I was using ownCloud (better communications via the web, faster bug fixes, more reliable).

I switched to this android application and for now it is working for me.
[https://github.com/stefan-niedermann/nextcloud-notes](https://github.com/stefan-niedermann/nextcloud-notes)
[https://f-droid.org/repository/browse/?fdid=it.niedermann.owncloud.notes](https://f-droid.org/repository/browse/?fdid=it.niedermann.owncloud.notes)

**Comment by ramschmaerchen** [comment]
Great, good to know.

**Comment by 9662** [comment]
> OwnCloud was never known for stable releases. Bugfixing has always been a problem in the owncloud world. Seeing all those that were unable to maintain ownCloud in an enterprise way depart to nextCloud does not want me to use nextCloud.

Every project makes a (deliberate or not) decision on its stability vs innovation trade-off. Your comment appears to suggest that the OC/NC maintainers drift a bit too far towards the innovation side. At least from my experience and considering the "stable" release channel, that indeed appears to be a valid criticism for a project which is "enterprise-oriented"—let us hope that as the project evolves (it is still relatively young, after all) it will stabilise at a more central or conservative position.

> Calling a gsoc attender dead is just not honest.

GSOC acceptance is not a mark of quality per se and in any case in the opinion of @jancborchardt acceptance into the programme might have been influenced by work done by individuals who are no longer directly involved in the ownCloud project. I have referred to it as being dead in a previous comment on account of its current lack of effective leadership and direction.

> Fully free != Shipping with GCM per default.

That I fully agree and I was very surprised myself to see that having crept in! I am aware of Google's aggressive efforts to "close" Android, including via tactics which are frankly underhanded, such as making functionality available only via proprietary interfaces or walled gardens. Those efforts should be vigorously resisted, including in some cases via holding back features which rely on them until open alternatives are available. In the long run, we all lose and innovation stagnates if we let critical aspects of technology be controlled by a single actor, cf. Microsoft in the 1980-1990s.

In this respect, incidentally, I really appreciate the zeal of the [F-Droid](https://gitlab.com/fdroid) gang in keeping things open.

> We will look into supporting nextCloud anyways.

Thank you for your efforts! I have been using myOwnNotes for a few years now and I really appreciate its simple and predictable design, getting things done reliably and without unnecessary frills. While there are other note-taking Nextcloud apps, yours fills a unique niche and is currently irreplaceable.

**Comment by AndyScherzinger** [comment]
> Fully free != Shipping with GCM per default.

True and like @mario already comment, is something we are working on getting rid again. The reason why it came in and then it will get very tricky is that Android 7 broke large parts of the app for many Android 7 running devices. So this is a quick fix to enable people to use it again since a pure-no-goole fix would have taken much longer, so it was a temporary trade-off/quick fix. 
tl;dr we are working on getting it removed while keeping the feature intact

> Regarding myownnotes support it depends on the effort needed. We will look into it.

Accessing the accounts should work the exact same way for both apps, so it shouldn't be much effort

**Comment by kuffnukk** [comment]
Mee too

"I am forced to use the old version of MyOwnNotes until Nextcloud app is supported"

But I using the Notes app from Niedermann IT, too.

**Comment by Albirew** [comment]
A "simple" workaround would be to add a button "I want to configure credentials myself" leading to a page with fields for server address and credentials so people wouldn't have to depend to cloud app to use MyOwnNotes.

**Comment by ramschmaerchen** [comment]
The ownCloud application dependency is on purpose and was a requested feature of Version 1.x

**Comment by kuffnukk** [comment]
I using the Notes app from Niedermann IT some weeks now. (Made in Germany, as I ;-) Clean GUI and some nice features such as MarkDown. Works with ownCloud- & Nextcloud-Server Notes app. https://github.com/stefan-niedermann/nextcloud-notes
Available at F-droid.org & Google-PlayStore

**Comment by ramschmaerchen** [comment]
Good for you

**Comment by thomascrisan** [comment]
I would like to add something; Thank you to the developers of MyOwnNotes.  I used your application for years and I greatly appreciated your work.  It is just for me, the last revision was a non-starter and I hope you can address the issues moving forward.

BTW, I don't use either the ownCloud or Nextcloud sync applications.  I have found [FolderSync](http://www.tacit.dk/foldersync) meets my needs better than either of the own/Next cloud applications.

**Comment by tobiasKaminsky** [comment]
I'll add a PR later this day which supports also nextcloud, I hope it will be merged.

**Comment by tobiasKaminsky** [comment]
Here it is : #149

**Comment by ramschmaerchen** [comment]
Thanks. Will merge after testing.

**Comment by jospoortvliet** [comment]
Any progress? I had some ppl ask me about it ;-)

WRT your comments about stability - feel free to just look a bit on twitter and read about how people had fun with ownCloud 10.0.0 and compare it with Nextcloud 12.0.0 - that should be all you need to be convinced which project has a focus on stability. Software will never be completely stable, of course and certainly there are and will be bugs in Nextcloud but we'll work hard to make it more stable, promise.

**Comment by kdslkdsaldsal** [comment]
With the danger of this issue getting even more off-topic but this needs a comment:

> WRT your comments about stability - feel free to just look a bit on twitter and read about how people had fun with ownCloud 10.0.0 and compare it with Nextcloud 12.0.0 - that should be all you need to be convinced which project has a focus on stability. Software will never be completely stable, of course and certainly there are and will be bugs in Nextcloud but we'll work hard to make it more stable, promise.

So much about stability in nC: [1](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/2) [2](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/5l) [3](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/6) [4](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/10) [5](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/12) [6](https://help.nextcloud.com/t/welcome-to-nextcloud-12/12874/17) (I just had a very short look at the Nc 12 annoucement topic for this).

Sure, both projects have their issues (especially with the first versions of a major release like oC 10 and nC 12). But both projects are trying their best to get their releases stable as the user is expecting this.

That would have been also a way better answer to the mentioned "stability" then just the FUD from above.

**Comment by jancborchardt** [comment]
Ah wow, the @kdslkdsaldsal fake account at it again.

I'll unsubscribe here. We offered our help and implemented things instead of just talking, as seen above. :) How you continue with it with your app is up to you of course.

**Comment by kdslkdsaldsal** [comment]
Wow, havn't expected anything else then just accuses without seeing the point of my comment.

Just to keep this at the same same language / level of the previous comment:

Yes, i care about something like i have quoted above as long as a Nextcloud employee spreading fake facts  to just to push the own product.

I'll unsubscribe here as well, have fun living in your own bubble around Nextcloud.

**Comment by jancborchardt** [comment]
> Yes, i care about something like i have quoted above as long as a Nextcloud employee spreading fake facts to just to push the own product.

Let me make two things very clear:

1. I always cared about this app and other 3rdparty apps, when I was at OwnCloud (even when there was no company) and also now that I'm at Nextcloud. That's very apparent from my past issue reports and participation here.

2. At least I don't veil the fact where I'm employed. Acting as if you are an independent party is just ridiculous.

**Comment by kdslkdsaldsal** [comment]
1. You might have noticed that my previous comments are about https://github.com/aykit/MyOwnNotes/issues/140#issuecomment-304528620, and not about any of your activity here (besides the "Fake-Account").

2. I can't vile something which doesn't exist. Sure i'm not independent, just because i don't like how Nextcloud people acting against ownCloud publicly (best example https://github.com/aykit/MyOwnNotes/issues/140#issuecomment-304528620)

**Comment by jhf2442** [comment]
Back to topic... any news on the PR being merged and released ?
Downloaded the client today from F-Droid and still got a version that's depending on the OC accounts being configured

**Comment by tobiasKaminsky** [comment]
@jhf2442 there is a PR by pending, which can solve this #149

**Comment by jhf2442** [comment]
Hi Tobias, that's what I was talking about... any ETA for it ? (though you are the submitter and not the approver)

**Comment by tobiasKaminsky** [comment]
Sorry, I misread it.
No ETA from my side, but I am here and willing to change anything that comes on their QA side to get this merged as soon as possible.

**Comment by makayabou** [comment]
Hello,
I don't really understand what is the problem with the feature of supporting login with nextcloud app, but I just want to say that I'd really love to see that happen!!
Thanks folks for all the great work you do that I couldn't..

**Comment by maltefiala** [comment]
Releases and tags have been removed. Will be re-released as soon as master is working again. Sorry for the trouble.

**Comment by twixi** [comment]
please!! need support nextcloud!

**Comment by BuxarNET** [comment]
need support nextcloud!
